stacks-signer/src/runloop.rs

```diff
@@ -0,0 +1,439 @@
+use frost_signer::{
+    config::PublicKeys,
+    net::Message,
+    signing_round::{MessageTypes, Signable, SigningRound},
+};
+use libsigner::{SignerRunLoop, StackerDBChunksEvent};
+use p256k1::ecdsa;
+use slog::{slog_info, slog_warn};
+use stacks_common::{info, warn};
+use std::time::Duration;
+use wsts::Point;
+
+use crate::{
+    config::Config,
+    crypto::{frost::Coordinator as FrostCoordinator, Coordinatable, OperationResult},
+    stacks_client::StacksClient,
+};
+
+/// Which operation to perform
+#[derive(PartialEq, Clone)]
+pub enum RunLoopCommand {
+    /// Wait for an initial command
+    Wait,
+    /// Continaully poll for commands to execute
+    Run,
+    /// Generate a DKG aggregate public key
+    Dkg,
+    /// Generate a DKG aggregate public key and then sign the message
+    DkgSign {
+        /// The bytes to sign
+        message: Vec<u8>,
```

Comment:
Presumably, the `message` will be decoded and validated by the business logic before signing, right?  For example, the signer might want to check that the data to sign is a partially-signed Bitcoin transaction, or a well-formed Stacks block, but nothing else? If so, then would it make sense to treat this field as an `enum` of the possible types of data that could be signed, so only valid inputs can be represented?

Reply:
Presumably there will be a small set of things we are willing to sign, and they will definitely need to be validated before doing so.  There's lots of weird crypto things that can happen if you can trick people into signing arbitrary messages.

It's definitely something to think about but seems very orthogonal to this PR itself.

Reply:
> There's lots of weird crypto things that can happen if you can trick people into signing arbitrary messages.

That's what I'm worried about.

> It's definitely something to think about but seems very orthogonal to this PR itself.

Either way, it seems important that we don't forget to do this.  Can you put this into an issue before merging?

Reply:
Issue created https://github.com/stacks-network/stacks-blockchain/issues/3926